joblib/test/test_numpy_pickle.py

```diff
@@ -245,15 +316,77 @@ def test_masked_array_persistence():
     nose.tools.assert_true(isinstance(b, np.ma.masked_array))
 
 
-def test_z_file():
-    # Test saving and loading data with Zfiles
+@with_numpy
+def test_compress_mmap_mode_warning():
+    # Test the warning in case of compress + mmap_mode
+    rnd = np.random.RandomState(0)
+    a = rnd.random_sample(10)
+    this_filename = env['filename'] + str(random.randint(0, 1000))
+    numpy_pickle.dump(a, this_filename, compress=1)
+    with warnings.catch_warnings(record=True) as caught_warnings:
+        warnings.simplefilter("always")
+        numpy_pickle.load(this_filename, mmap_mode='r+')
+        nose.tools.assert_equal(len(caught_warnings), 1)
+        for warn in caught_warnings:
+            nose.tools.assert_equal(warn.category, DeprecationWarning)
+            nose.tools.assert_equal(warn.message.args[0],
+                                    'File "%(filename)s" appears to be '
+                                    'buffered, this is not compatible with '
+                                    'mmap_mode "%(mmap_mode)s" flag passed' %
+                                    {'filename': this_filename,
+                                     'mmap_mode': 'r+'})
+
+
+@with_numpy
+def test_cache_size_warning():
+    # Check deprecation warning raised when cache size is not None
     filename = env['filename'] + str(random.randint(0, 1000))
-    data = numpy_pickle.asbytes('Foo, \n Bar, baz, \n\nfoobar')
-    with open(filename, 'wb') as f:
-        numpy_pickle.write_zfile(f, data)
-    with open(filename, 'rb') as f:
-        data_read = numpy_pickle.read_zfile(f)
-    nose.tools.assert_equal(data, data_read)
+    rnd = np.random.RandomState(0)
+    a = rnd.random_sample((10, 2))
+
+    for cache_size in (None, 0, 10):
+        with warnings.catch_warnings(record=True) as caught_warnings:
+            warnings.simplefilter("always")
+            numpy_pickle.dump(a, filename, cache_size=cache_size)
+            expected_nb_warnings = 1 if cache_size is not None else 0
+            nose.tools.assert_equal(len(caught_warnings),
+                                    expected_nb_warnings)
+            for warn in caught_warnings:
+                nose.tools.assert_equal(warn.category, DeprecationWarning)
+                nose.tools.assert_equal(warn.message.args[0],
+                                        "Please do not set 'cache_size' in "
+                                        "joblib.dump, this parameter has no "
+                                        "effect and will be removed. "
+                                        "You used 'cache_size={0}'".format(
+                                            cache_size))
+
+
+@with_numpy
+@with_memory_usage
+def test_memory_usage():
+    """Verify memory stays within expected bounds."""
```

Comment:
I tend to stay clear of using docstring for test functions. The reason: nosetests use the docstring rather than the name of the function in case of failure.

```
======================================================================
ERROR: Test if a known example file gets properly split
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/volatile/le243287/miniconda3/lib/python3.4/site-packages/nose/case.py", line 198, in runTest
```

In general the first thing I want to do in case of failure is to run `nosetests -m` to only run the failing test. `nosetests -m` is based on the name of the test function and I find it mildly annoying to do a `git grep -p' to figure out the name of the function from the docstring.